conftest.py

```diff
@@ -177,8 +178,10 @@ def device_mesh(request, silicon_arch_name, silicon_arch_wormhole_b0, device_par
     if isinstance(param, tuple):
         grid_dims = param
         assert len(grid_dims) == 2, "Device mesh grid shape should have exactly two elements."
-        device_grid = ttnn.DeviceGrid(*grid_dims)
         num_devices_requested = grid_dims[0] * grid_dims[1]
+        if num_devices_requested > len(device_ids):
+            pytest.skip("Requested more devices than available. Test not applicable for machine")
```

Comment:
Do we wanna skip here? Don't we want to know if the machine is not properly set-up or if the wrong one is requested / assigned?

Reply:
I wasn't sure if there was a better place for this but previously tests would filter themselves based on len(devices) but now with `device_mesh` the test doesn't get the chance to query this attribute. I can see that this could negatively impact the CI infra from diagnosing a machine issue from a test issue so I can change how this is done.

Reply:
Now that I think about it, I think we should probably have more defensive barriers at set up time for this, which is a lower priority item.

I think this is fine, but how are test writers expected to specify the grid they need? And is this enforced? Just wondering for consistency among the pytests.

Reply:
The device grid request is  specified like this:
```
@pytest.mark.parametrize("device_mesh", [pytest.param((8, 4), id="8x4_grid")], indirect=True)
```
and then the test author could do something like"

```
    if len(device_mesh.get_devices()) != 32:
        pytest.skip("Not TG!")
```

Reply:
Hmm but just to be clear with these proposed changes, your example would skip in `device_mesh` fixture now rather than at the skip that the author put in within the test?

Reply:
The reason I'm skipping here is that right after this skip, we assert if there aren't enough chips. However, the test has no way to know that when it invokes the fixture. 

I didn't actually explicitly add my new test file to any test lists (I believe it's picked up and run automatically). I can filter it out at the pytest script invocation instead if that's an option.